components/Cards/DetailsCard.js

```diff
@@ -0,0 +1,180 @@
+import { StyledCardSubHeading } from "./Card";
+
+export default function DetailsCard({ id, isFavorite, toggleFavorite, place }) {
+  const {
+    _id,
+    name,
+    description,
+    region,
+    image,
+    temperature,
+    reviews,
+    activities,
+  } = place;
+
+  const [comments, setComments] = useState([]);
+
+  // Define fetchComments function outside the useEffect
+  const fetchComments = async () => {
+    try {
+      const response = await fetch(`/api/comments?placeId=${_id}`);
+      if (response.ok) {
+        const data = await response.json();
+        setComments(data);
+      } else {
+        console.error("Failed to fetch comments");
+      }
+    } catch (error) {
+      console.error("Error fetching comments:", error);
+    }
+  };
+
+  useEffect(() => {
+    // Fetch comments associated with the current place
+    fetchComments();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [_id]);
+
+  return (
+    <>
+      <StyledCardArticle key={_id}>
+        <StyledCardImage
+          src={image}
+          sizes="100vw"
+          style={{
+            width: "100%",
+            height: "auto",
+          }}
+          width="0"
+          height="0"
+          alt={name}
+          loading="lazy"
+        />
+        <StyledCardBody>
+          <StyledCardTitle>{name}</StyledCardTitle>
+          <FavoriteButton
+            id={_id}
+            isFavorite={isFavorite}
+            toggleFavorite={toggleFavorite}
+          />
+          <StyledCardSubHeading>
+            {region} &#183; &nbsp;
+            {activities.map((activity) => activity.activityname).join(", ")}
+          </StyledCardSubHeading>
+          <StyledCardSubHeadings>Description:</StyledCardSubHeadings>
+          <StyledCardText>{description}</StyledCardText>
+          <StyledCardSubHeadings>Reviews:</StyledCardSubHeadings>
+          <StyledCardText>
+            {reviews.map((review) => review.review).join(", ")}
+          </StyledCardText>
+        </StyledCardBody>
+      </StyledCardArticle>
+      <FormComments fetchComments={fetchComments} />
+      {comments.length > 0 && (
+        <StyledComments>
+          <h3>Comments:</h3>
+          {comments
+            .map((comment) => (
+              <div key={comment._id}>
+                <StyledCommentsAnswers>
+                  {!comment.userImage ? (
+                    <StyledCommentsImage
+                      src="/images/profile-placeholder.png"
+                      width="50"
+                      height="50"
+                      alt="Profile picture"
+                      loading="lazy"
+                    />
+                  ) : (
+                    <StyledCommentsImage
+                      src={comment.userImage}
+                      width="50"
+                      height="50"
+                      alt="Profile picture"
+                      loading="lazy"
+                    />
+                  )}
```

Comment:
This looks a lot more readable to me:

```suggestion
                    <StyledCommentsImage
                      src={comment.userImage ? comment.userImage : "/images/profile-placeholder.png"}
                      width="50"
                      height="50"
                      alt="Profile picture"
                      loading="lazy"
                    />
```